### ISSUE TITLE
Problems with changing admin URL after update to 12.1.2

### ISSUE THREAD

**Issue by vinny1962** [pH7Software/pH7-Social-Dating-CMS#191]
I updated from 12.0.0 to 12.1.2 today I am fairly new to this software although I had this working before in 12.0.0. I was able to go into the protected files and search for admin123 under all. The admin123 folder would appear with another admin folder to the right that would have number after the admin123. According to documentation you are suppose to rename the admin123 folder to what you want for security purposes. After this is changed you have to go into protected files /YOURS_PROTECTED_FOLDER/app/configs/constants.php and change the define ('PH7_ADMIN_MOD', ' admin123 '); I tried this 7 times. Of course you have to rename the folder to what you want then go into your hosting panel and search in the _protected/app/configs/constants.php and change it there after you change the folder. What seems to be happening is if I do not get the error it works how ever I am getting a error message is stating your _protected file is not there when I try to go into the page modules.tpl   I found this really strange I have been pulling my hair all day.

**Comment by pH-7** [comment]
Hiya!

No sure why you really want to move to 12.1.2 since the changes for you aren't big. 

Anyway, this might help you.

1. Please run the following through your phpmyadmin on pH7CMS database.
Also, replace 'ph7_' by your prefix set in "~/_protected/app/configs/config.ini" [database] prefix

```sql
INSERT INTO ph7_settings (settingName, settingValue, description, settingGroup) VALUES
('backgroundColor', '', 'Overwrite background color', 'design'),
('textColor', '', 'Overwrite text color', 'design'),
('footerLinkColor', '', 'Overwrite footer links color', 'design'),
('linkColor', '', 'Overwrite links color', 'design'),
('linkHoverColor', '', 'Overwrite links hover color', 'design');

UPDATE ph7_modules SET version = '1.4.3' WHERE vendorName = 'pH7CMS';
```

2. Secondly, please remove all subfolder in `_protected/data/cache/`

P.S. And make sure all files are replaced by the new ones.

P.S.2 pH7CMS 12.1 caches all dynamic URLs generated from pH7Framework's router. Maybe... that's why you had some issues there...

**Comment by vinny1962** [comment]
I am not sure if this is the problem I did a complete new install using
Softaculous I delete the old account off the server prior to backing up I
have a complete admin backup from the server of 12.0.0 I updated because I
want new features I should not have to do anything to the database and a
maybe could lead into another problem.

**Comment by polynamaude** [comment]
If you are trying to use the database without rebuilding by the installation script but only upgrading the files. Then you'll need to run the script @pH-7 gave you.
You should try clearing the cache like indicated in the docs by removing files from some of the cache folder. 
This comment may apply https://github.com/pH7Software/pH7-Social-Dating-CMS/issues/154#issuecomment-351591753

And the documentation (clearing the cache)
http://ph7cms.com/doc/

Good luck !

PS: Your question is a bit hard to follow in the way you describe. Maybe you should make it more concise in the details you give so we could help you better. Also I'd recommend installing directly the software and not by using a script like "Softaculous" as we never know how does this system drive the installation and it gives us one more possibility of problem (and options in differential diagnosis).

As you don't seem to have a live system installed maybe I could suggest that you start over with the latest release and don't bother trying to go through the upgrade procedures.

**Comment by vinny1962** [comment]
I did not go through the upgrade process at all as I said I terminated the
account off my server completely and created a new account I then installed
PH7CMS. I see allot of issues and I do not believe its related to
Softaculous I can not change the admin123 folder as I was able to in
version 12.0.0 installed on march 17th this month all was working well. I
also noticed this web site color changer is not working proper also I
tested this software on two separate domains and its doing the same thing
on both. At the footer you have five menu links correct. I have noticed
that on both installs of this PH7CMS the about us link is red and the help
link is red by default on hover very strange. All other links are pink on
hover

When I tried to change the link color for the footer it only change the
(newsletter) (invite) (sitemap) the links coming off the about us menu did
change although the link stayed the same color about us will not change.


Regards Vinny

**Comment by polynamaude** [comment]
I understand nearly nothing from your messages, so I'll simply quote what you said and ask questions.

**_I did not go through the upgrade process at all (OK) as I said I terminated the account off my server completely_**
What account kind of account or you talking about ?? To install a web application you simply create a folder, make it visible on your webserver and point to it in your browser to start the process
**_and created a new account I then installed PH7CMS._**
Created what type of account ? A new shell user ? A new hosting account with a provider ?

**_I can not change the admin123 folder as I was able to in version 12.0.0 installed on march 17th this month all was working well._** 

Everything was working well ? Would have been nice to say thanks to the helpings hands.
How do you proceed to change the folder ? The way you describe your actions is incomprehensible. You open a shell, go to the folder using **cd** and after you change the name of the admin123 folder to what you want for example
**cd protected
cd app
cd system
cd modules
mv admin123** _mysecretadmin_
after you modify the constant.php file :
( from your last acions )
**cd ..
cd ..
cd configs
nano constants.php**
**_I tested this software on two separate domains and its doing the same thing on both_**

Your affirmation about trying to install on two different domains doesn't have any link or use of understanding. You could install a software of thousands of domain and always end up having the same problems if you have a configuration problem that you repeat. There's no link between a domain name having something to do with software configuration. Many CMS (like this one) can easily be changed of domain name by reconfiguration if needed throughout the cycle of operation.

_**When I tried to change the link color for the footer it only change the (newsletter) (invite) (sitemap) the links coming off the about us menu did change although the link stayed the same color about us will not change.**_ 

How did you act to change the links ? You hand edited the .tpl file ? You used the included file editor ? Used your hosting provider interface to do edition ? Maybe to describe the actions you did in a simple way could help out !
We're now into many issue so I suggest simply we start by solving one at a time. If you do consider that you problem is not softaculous - related then what would it cost you to follow the instruction given in the installation instructions ? You're not the only user of this software but as you're having problem it would be a wise move to do as recommended. It also means to follow advice that we give you when suggesting to try something out and giving feedback.

Professional software manager won't do upgrade on live system and always test them on a testbed before implanting as we never know what situation you may encounter. I'll also suggest you give some information about what you're actually using (Apache version, list of modules activated).

I'll be quite honest here, I think you're more busy arguing that you know where the problem is and push it onto our side than to do the manipulations we suggest to help out. No one has the obligation to give you technical support so the least you can do is collaborate kindly with people who give there time helping out. As it is going on, I'll make this probably my last intervention for your case as not only you don't seem to follow suggestions but at the same time I'm getting doubt about the knowledge you got to do command line based manipulation or in-depth understanding of web application further than being limited to "point-click-next".

Helping is a participative action that require oneself involvement of the person making the request.

Good luck !

**Comment by vinny1962** [comment]
I do not know what you do not understand. I terminated the account from a
cPanel hosting server that version 12.0.0 was on the domain matchudate.com
that account was terminated leaving no more data no more anything. I
created a new cPanel account and I did the install of PH7CMS 12.1.2 . This
is the link to the documentation I have used for the admin123 change to
rename the admin folder.  https://ph7cms.com/doc/ It is number 21 down the
list renaming the admin folder. This worked for me on the version 12.0.0
although when I did the version 12.1.2 there was problems with the renaming
the admin folder I was getting errors that stated your _protected file no
longer exist when I went to the page modules links such as aboutus.tpl and
so on.

Sincerely Yours
Vinny

**Comment by polynamaude** [comment]
Got that right about where you've got your information but you're not telling me which way you did for renaming your folder ?

I won't repeat myself again saying to send screenshot, logs or other explanation of what's going on.

Good luck ! I'm gone

**Comment by pH-7** [comment]
I wasn't aware you are running it through Softaculous! Then, things are different (same for the custom admin123 folder. Softaculous doesn't know the change there and that might be the problem!). 
It might have an issue from they side with the update process. Like I said to you by email, it would be wiser to wait until a next release (**or install it from [Sourceforge](https://sourceforge.net/projects/ph7socialdating/files/latest/download) without using Softaculous if you still get an issue** with it). 
There is no hurry to update from 12.0 to 12.1 for the moment. Just wait and enjoy with what you already have :smiley:

**Update from Softaculous won't work if you renamed a folder**

**Comment by pH-7** [comment]
@polynamaude You can rename the admin folder (so, people won't be able to see the admin login page).  https://ph7cms.com/doc/en/rename-the-admin-folder

**Comment by pH-7** [comment]
@vinny1962 I couldn't replicate your issue with renamed admin folder. Please follow our advice. And maybe ditch Softaculous for the moment... We cannot do more for the moment unfortunately. If you get another problem, please just reopen a new issue.

Have a great weekend! 🎉

**Comment by polynamaude** [comment]
@pH-7 I'm not sure why you're saying that to me here ? I certainly know that it's possible to rename the folder and if you look through the messages I wrote, I've explained to him how to do so using the command line shell and using nano to modify the constants.php file.

And I gave the same suggestion as you as to install using plain gitHub / Sourceforge as it's always hard to know if the problem is coming from softaculous or if it's on the CMS side. The simpler you go with a solution, the easier it is to find out where the problem reside. One step at a time.

I've read to the documentation and found that the idea was really nice to point out. You can also do the same with other CMS like changing the "administrator" in Joomla but it does require some more work and it's not described in the documentation.
